renderdoc/driver/gl/wrappers/gl_draw_funcs.cpp

```diff
@@ -709,10 +709,47 @@ bool WrappedOpenGL::Serialise_glDrawArrays(GLenum mode, GLint first, GLsizei cou
   return true;
 }
 
+void WrappedOpenGL::CopyClientMemoryArrays(GLint first, GLsizei count)
+{
+  GLuint prevBinding;
+  m_Real.glGetIntegerv(eGL_ARRAY_BUFFER_BINDING, (GLint *)&prevBinding);
+
+  for(GLuint i = 0; i < ARRAY_COUNT(m_ClientMemoryVBOs); i++)
+  {
+    GLint enabled, size, stride;
+
+    m_Real.glGetVertexAttribiv(i, eGL_VERTEX_ATTRIB_ARRAY_ENABLED, &enabled);
+    if(!enabled)
+      continue;
+
+    // Check that the active buffer is one of the client-memory copy buffers.
+    GLuint buffer;
+    m_Real.glGetVertexAttribiv(i, eGL_VERTEX_ATTRIB_ARRAY_BUFFER_BINDING, (GLint *)&buffer);
+    if(buffer < m_ClientMemoryVBOs[0] ||
+       buffer > m_ClientMemoryVBOs[ARRAY_COUNT(m_ClientMemoryVBOs) - 1])
```

Comment:
Potential edge-case - I don't think `glGenBuffers` guarantees that it will return a contiguous set of IDs. Imagine the case where someone creates a context, allocates and deletes some buffers leaving the name space sparse, and then creates a new context where most of the work will happen - our VBO names are then non-contiguous.

If you use the ResourceIds for the VBOs, those should be safely contiguous.